src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java

```diff
@@ -233,150 +232,30 @@ private void processRequest(HttpServletRequest request, HttpServletResponse resp
                         "Sorry. OAI Service is disabled on this Dataverse node.");
                 return;
             }
+                        
+            RawRequest rawRequest = RequestBuilder.buildRawRequest(httpServletRequest.getParameterMap());
             
-            OAIRequestParametersBuilder parametersBuilder = newXoaiRequest();
-            
-            for (Object p : request.getParameterMap().keySet()) {
-                String parameterName = (String)p; 
-                String parameterValue = request.getParameter(parameterName);
-                parametersBuilder = parametersBuilder.with(parameterName, parameterValue);
-
-            }
-            
-            OAIPMH handle = dataProvider.handle(parametersBuilder);
+            OAIPMH handle = dataProvider.handle(rawRequest);
             response.setContentType("text/xml;charset=UTF-8");
-            
-            if (isGetRecord(request) && !handle.hasErrors()) {
-                writeGetRecord(response, handle);
-            } else if (isListRecords(request) && !handle.hasErrors()) {
-                writeListRecords(response, handle);
-            } else {
-                XmlWriter xmlWriter = new XmlWriter(response.getOutputStream());
-                xmlWriter.write(handle);
-                xmlWriter.flush();
-                xmlWriter.close();
-            }
+
+            XmlWriter xmlWriter = new XmlWriter(response.getOutputStream(), repositoryConfiguration);
+            xmlWriter.write(handle);
+            xmlWriter.flush();
+            xmlWriter.close();
                        
         } catch (IOException ex) {
             logger.warning("IO exception in Get; "+ex.getMessage());
             throw new ServletException ("IO Exception in Get", ex);
-        } catch (OAIException oex) {
-            logger.warning("OAI exception in Get; "+oex.getMessage());
-            throw new ServletException ("OAI Exception in Get", oex);
         } catch (XMLStreamException xse) {
             logger.warning("XML Stream exception in Get; "+xse.getMessage());
             throw new ServletException ("XML Stream Exception in Get", xse);
-        } catch (XmlWriteException xwe) {
-            logger.warning("XML Write exception in Get; "+xwe.getMessage());
-            throw new ServletException ("XML Write Exception in Get", xwe);  
         } catch (Exception e) {
             logger.warning("Unknown exception in Get; "+e.getMessage());
             throw new ServletException ("Unknown servlet exception in Get.", e);
         }
```

Comment:
**nitpick**: I'm not sure if this might be an Exception anti-pattern here.

1) Is the ServletException exposed to a user, which will include the pesky internal details? (That would be an anti-pattern)
2) Is the ServletException logged by the app server, too? In that case: would it be wise to avoid double logging the same thing? (Which would also be an anti-pattern)